docs/ForDevelopers/Dependencies.md

```diff
@@ -24,45 +29,115 @@ For other recent Linux distributions, you can also rely on the pre-built binarie
+cmake --build build --target install -j$(sysctl -n hw.logicalcpu)
+```
+</TabItem>
+</Tabs>
 
 ## Apache Arrow
-### Ubuntu/Linux
+
+<Tabs groupId="os-tabs">
+<TabItem value="linux" label="Ubuntu/Linux">
+
 ```shell
 wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
 apt install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename  --short).deb
 apt-get update
-apt-get install libarrow-dev=19.*
+apt-get install libarrow-dev=20.*
 ```
+
 ### Binaries
 For other recent Linux distributions, you can also rely on the pre-built binaries provided by the Apache Arrow project.
 
 ### Building from Source
 
 ```shell
-wget https://dlcdn.apache.org/arrow/arrow-19.0.1/apache-arrow-19.0.1.tar.gz
-tar -xf apache-arrow-19.0.1.tar.gz
-RUN cd apache-arrow-19.0.1/cpp
-cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=[output-dir] -DARROW_DEPENDENCY_SOURCE=BUNDLED -DARROW_BUILD_STATIC=ON -DARROW_CSV=ON -DARROW_COMPUTE=ON
+wget https://github.com/apache/arrow/releases/download/apache-arrow-20.0.0/apache-arrow-20.0.0.tar.gz
+tar -xf apache-arrow-20.0.0.tar.gz
+cd apache-arrow-20.0.0/cpp
+cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=[output-dir] -DARROW_DEPENDENCY_SOURCE=BUNDLED -DARROW_BUILD_STATIC=ON -DARROW_CSV=ON -DARROW_COMPUTE=ON apache-arrow-20.0.0/cpp
 cmake --build build --target install -j$(nproc)
 ```
+
+</TabItem>
+<TabItem value="macos" label="MacOS">
+
+Install Apache Arrow using Homebrew:
+
+```shell
+brew install apache-arrow
+```
+
+⚠️ **Caution**: Make sure this is the current version. The homebrew apache-arrow recipe currently does not support version pinning.
+
+### Building from Source
```

Comment:
In my environment, I encountered the following issue, which required replacing -G Ninja with -G "Unix Makefiles":
https://github.com/ninja-build/ninja/issues/1463

Additionally, I ran into an error indicating that Z3 was not installed. I resolved this by running brew install z3.
Would it be possible to include this command as part of the prerequisites for building LLVM from source?